### ISSUE TITLE
Tag prefix ignored during checkWorkingTree() call for lerna version and publish

### ISSUE THREAD

**Issue by treywashington** [lerna/lerna#2056]
Calling `lerna version` on a tagged HEAD commit errors regardless of whether the tag matches `--tag-version-prefix`:
 ```
ERR! ERELEASED The current commit has already been released. Please make new commits before continuing.
```

## Expected Behavior

Expected behavior is that `lerna version` and `lerna publish` will version the commits since the last release tag matching the `--tag-version-prefix` set or default 'v'.

Example: 
```bash
> git describe --long // otherTagScheme.10.0.0-0-gc9a9d91
> git describe --match v* --long // v0.0.1-50-gc9a9d91
> lerna version patch // Changes: @scope/pkg: 0.0.1 => 0.0.2
```

## Current Behavior

Current behavior is to treat any tag returned from `git describe --long` as the tag since the last lerna release. 

Example:
```bash
> git describe --long // otherTagScheme.10.0.0-0-gc9a9d91
> git describe --match v* --long // v0.0.1-50-gc9a9d91
> lerna version patch // ERR! ERELEASED The current commit has already been released. Please make new commits before continuing.
```

## Possible Solution

Suggested fix is to pass through the match option to `describeRef()` in  [checkWorkingTree()](https://github.com/lerna/lerna/blob/86ba7a03fe8aa86733d4dd1de8136ee918fe698b/utils/check-working-tree/lib/check-working-tree.js#L13)

The match option can be derived from the execOption `--tag-version-prefix` for both the `publish` and `version` commands.

Corollary to this is to make sure that `lerna version` updates the packages changed since `git describe --match ${tag-version-prefix}* ` as well, not if its a tag that doesn't match. 

## Context

Context here is that my project has two different release cadences for two different repositories right now. We are trying to configure our npm repo deployment with lerna but having two different tagging schemes is throwing it off.

## Your Environment

| Executable | Version |
| ---: | :--- |
| `lerna --version` | 3.5.1 |
| `npm --version`  | 6.4.1 |
| `yarn --version` | 1.9.4 |
| `node --version` | 11.2.0 |

| OS | Version |
| --- | --- |
| macOS Mojave | 10.14.4 |

**Comment by evocateur** [comment]
Yeah, that is something we should definitely be doing. PRs welcome!
